src/library/scala/collection/immutable/ListSet.scala

```diff
@@ -119,12 +119,8 @@ sealed class ListSet[A] extends AbstractSet[A]
     var that: ListSet[A] = self
     def hasNext = that.nonEmpty
     def next: A =
-      if (hasNext) {
-        val res = that.head
-        that = that.tail
-        res
-      }
-      else Iterator.empty.next()
+      if (!hasNext) throw new NoSuchElementException("next on empty iterator")
```

Comment:
Revert to delegation to `Iterator.empty.next()`. This avoids duplicating exception messages.

      /** The iterator which produces no values. */
    val empty: Iterator[Nothing] = new AbstractIterator[Nothing] {
      def hasNext: Boolean = false
      def next(): Nothing = throw new NoSuchElementException("next on empty iterator")
    }

Delegating to an empty instance to trigger an exception is also seen in other places in the std. lib. for example from `Range`,

    final override def tail: Range = {
      if (isEmpty)
          Nil.tail

      drop(1)
    }

Here `Nil.tail` is used for it's side effect,

    override def tail: List[Nothing] =
      throw new UnsupportedOperationException("tail of empty list")